meta/serialize.pm

```diff
@@ -874,22 +875,22 @@ sub CreateDeserializeEmitMacros
     # provided then we need to release memory, actually only in places where we
     # have count and not const count
 
-    WriteSource "#define EXPECT(x) { \\";
-    WriteSource "    if (strncmp(buf, x, sizeof(x) - 1) == 0) { buf += sizeof(x) - 1; } \\";
-    WriteSource "    else { \\";
-    WriteSource "        SAI_META_LOG_WARN(\"expected '%s' but got '%.*s...'\", x, (int)sizeof(x), buf); \\";
-    WriteSource "        return SAI_SERIALIZE_ERROR; } }";
-    WriteSource "#define EXPECT_QUOTE     EXPECT(\"\\\"\")";
-    WriteSource "#define EXPECT_KEY(k)    EXPECT(\"\\\"\" k \"\\\":\")";
-    WriteSource "#define EXPECT_NEXT_KEY(k) { EXPECT(\",\"); EXPECT_KEY(k); }";
-    WriteSource "#define EXPECT_CHECK(expr, suffix) {                                 \\";
-    WriteSource "    ret = (expr);                                                  \\";
-    WriteSource "    if (ret < 0) {                                                 \\";
-    WriteSource "        SAI_META_LOG_WARN(\"failed to deserialize \" #suffix \"\");      \\";
-    WriteSource "        return SAI_SERIALIZE_ERROR; }                              \\";
-    WriteSource "    buf += ret; }";
-    WriteSource "#define EXPECT_QUOTE_CHECK(expr, suffix) {\\";
-    WriteSource "    EXPECT_QUOTE; EXPECT_CHECK(expr, suffix); EXPECT_QUOTE; }";
+    WriteHeader "#define EXPECT(x) { \\";
+    WriteHeader "    if (strncmp(buf, x, sizeof(x) - 1) == 0) { buf += sizeof(x) - 1; } \\";
+    WriteHeader "    else { \\";
+    WriteHeader "        SAI_META_LOG_WARN(\"expected '%s' but got '%.*s...'\", x, (int)sizeof(x), buf); \\";
+    WriteHeader "        return SAI_SERIALIZE_ERROR; } }";
+    WriteHeader "#define EXPECT_QUOTE     EXPECT(\"\\\"\")";
+    WriteHeader "#define EXPECT_KEY(k)    EXPECT(\"\\\"\" k \"\\\":\")";
+    WriteHeader "#define EXPECT_NEXT_KEY(k) { EXPECT(\",\"); EXPECT_KEY(k); }";
+    WriteHeader "#define EXPECT_CHECK(expr, suffix) {                                 \\";
+    WriteHeader "    ret = (expr);                                                  \\";
+    WriteHeader "    if (ret < 0) {                                                 \\";
+    WriteHeader "        SAI_META_LOG_WARN(\"failed to deserialize \" #suffix \"\");      \\";
+    WriteHeader "        return SAI_SERIALIZE_ERROR; }                              \\";
+    WriteHeader "    buf += ret; }";
+    WriteHeader "#define EXPECT_QUOTE_CHECK(expr, suffix) {\\";
+    WriteHeader "    EXPECT_QUOTE; EXPECT_CHECK(expr, suffix); EXPECT_QUOTE; }";
```

Comment:
move back to source

Reply:
The macros are also used by other c files such as saiserialize.c.  If they are moved back, then the same code have to be defined in other c files. Does this acceptable? Thanks!